test/statetest/statetest_runner.cpp

```diff
@@ -33,11 +34,23 @@ void run_state_test(const StateTransitionTest& test, evmc::VM& vm)
             state::finalize(state, rev, test.block.coinbase, 0, {});
 
             if (holds_alternative<state::TransactionReceipt>(res))
-                EXPECT_EQ(logs_hash(get<state::TransactionReceipt>(res).logs), expected.logs_hash);
+                if (!ignore_logs)
+                    EXPECT_EQ(
+                        logs_hash(get<state::TransactionReceipt>(res).logs), expected.logs_hash);
+                else
+                    EXPECT_TRUE(true);
             else
                 EXPECT_TRUE(expected.exception);
 
-            EXPECT_EQ(state::mpt_hash(state.get_accounts()), expected.state_hash);
+            if (!ignore_state_root)
+                EXPECT_EQ(state::mpt_hash(state.get_accounts()), expected.state_hash);
+            else
+            {
+                std::cerr << "{";
+                std::cerr << R"("stateRoot": ")" << state::mpt_hash(state.get_accounts()) << "\"";
+                std::cerr << "}\n";
```

Comment:
If you move these lines up before the clause, so they are emitted regardless .. then I don't see why `ignore_state_root` is even needed. The `jsonl` outputs are emitted on `stderr`, and the complaints about stateroot are emitted on `stdout`, so the caller can just ignore `stdout`. No need for evmone to do anything about it.